sdk/translation/azure-ai-translation-document/src/main/java/com/azure/ai/translation/document/models/TranslationSource.java

```diff
@@ -0,0 +1,271 @@
+     */
+    public String getPrefix() {
+        if (filter != null) {
+            prefix = filter.getPrefix();
+        }
+        return prefix;
+    }
+
+    /**
+     * Set the prefix property: A case-sensitive prefix string to filter documents
+     * in the source path for
+     * translation.
+     * For example, when using a Azure storage blob Uri, use the prefix
+     * to restrict sub folders for translation.
+     *
+     * @param prefix the prefix value to set.
+     * @return the TranslationSource object itself.
+     */
+    public TranslationSource setPrefix(String prefix) {
+        this.prefix = prefix;
+        if (filter == null) {
+            filter = new DocumentFilter();
+        }
+        filter.setPrefix(prefix);
+        return this;
+    }
+
+    /**
+     * Get the suffix property: A case-sensitive suffix string to filter documents
+     * in the source path for
+     * translation.
+     * This is most often use for file extensions.
+     *
+     * @return the suffix value.
+     */
+    public String getSuffix() {
+        if (filter != null) {
+            suffix = filter.getSuffix();
+        }
+        return suffix;
+    }
+
+    /**
+     * Set the suffix property: A case-sensitive suffix string to filter documents
+     * in the source path for
+     * translation.
+     * This is most often use for file extensions.
+     *
+     * @param suffix the suffix value to set.
+     * @return the TranslationSource object itself.
+     */
+    public TranslationSource setSuffix(String suffix) {
+        this.suffix = suffix;
+        if (filter == null) {
+            filter = new DocumentFilter();
+        }
+        filter.setSuffix(suffix);
+        return this;
+    }
+
+    /*
+     * Language code
+     * If none is specified, we will perform auto detect on the document
+     */
+    @Generated
+    private String language;
+
+    /*
+     * Storage Source
+     */
+    @Generated
+    private TranslationStorageSource storageSource;
+
+    /**
+     * Creates an instance of TranslationSource class.
+     *
+     * @param sourceUrl the sourceUrl value to set.
+     */
+    @Generated
+    public TranslationSource(String sourceUrl) {
+        this.sourceUrl = sourceUrl;
+    }
+
+    /**
+     * Get the sourceUrl property: Location of the folder / container or single file with your documents.
+     *
+     * @return the sourceUrl value.
+     */
+    @Generated
+    public String getSourceUrl() {
+        return this.sourceUrl;
+    }
+
+    /**
+     * Get the filter property: Document filter.
+     *
+     * @return the filter value.
+     */
+    @Generated
+    public DocumentFilter getFilter() {
+        return this.filter;
+    }
+
+    /**
+     * Set the filter property: Document filter.
+     *
+     * @param filter the filter value to set.
+     * @return the TranslationSource object itself.
+     */
+    @Generated
+    public TranslationSource setFilter(DocumentFilter filter) {
```

Comment:
See comment in [API view](https://apiview.dev/Assemblies/Review/2f419a8fe3fb47638c9d23db39f2c3b8?revisionId=a73043053d5b4d55a6ec5fd7e9d72381#com.azure.ai.translation.document.models.TranslationSource) about this.

Reply:
Decided to keep DocumentFilter class while we understand why the prefix and suffix were part of TranslationSource object in dotnet